lib/Sema/TypeCheckPattern.cpp

```diff
@@ -94,8 +94,6 @@ filterForEnumElement(DeclContext *DC, SourceLoc UseLoc,
     }
 
     if (auto *oe = dyn_cast<EnumElementDecl>(e)) {
-      // Ambiguities should be ruled out by parsing.
-      assert(!foundElement && "ambiguity in enum case name lookup?!");
```

Comment:
We shouldn't just remove this assertion. This amounts to a non-trivial behavior change in this function - it's now going to break ambiguities by selecting the last returned result from a lookup. The trouble here isn't the assert, it's the 

```
    if (e->isInvalid()) {
      continue;
    }
```

check guarding the filtering behavior. That condition was correct in 2016 when this was written, but is not great now that we've stabilized the computation of interface types and defined `isInvalid` in terms of it.

Reply:
I *think* this means this code just needs to become tolerant of duplicates. Quite what that means exactly I'm not sure - I need to go page all of this back in.

Reply:
Does that mean I might need to check for redeclarations in TypeCheckPattern?

Reply:
Thank you so much for helping me. And [Ramadan Kareem](https://www.thesun.co.uk/news/6211919/ramadan-kareem-mubarak-fasting-meaning/)! 🙏 :)